File: src/app/types/meme.ts
export type Meme = {
  id: string;
  title: string;
  top_text: string;
  bottom_text: string;
  template: string;
  author: string;
  votes: number;
  created_at: number;
};

export type MemeTemplate = {
  id: string;
  name: string;
  emoji: string;
  bgColor: string;
  textColor: string;
  accentColor: string;
  style: "classic" | "modern" | "chaos";
};

File: mobile/lib/models/meme.dart
class Meme {
  final String id;
  final String title;
  final String topText;
  final String bottomText;
  final String template;
  final String author;
  final int votes;
  final int createdAt;

  Meme({
    required this.id,
    required this.title,
    required this.topText,
    required this.bottomText,
    required this.template,
    required this.author,
    required this.votes,
    required this.createdAt,
  });

  factory Meme.fromJson(Map<String, dynamic> json) {
    return Meme(
      id: json['id'],
      title: json['title'],
      topText: json['top_text'] ?? '',
      bottomText: json['bottom_text'] ?? '',
      template: json['template'],
      author: json['author'],
      votes: json['votes'] ?? 0,
      createdAt: json['created_at'] ?? 0,
    );
  }

  DateTime get createdDate =>
      DateTime.fromMillisecondsSinceEpoch(createdAt * 1000);
}
